internal/services/report_processor.go

```diff
@@ -172,13 +174,13 @@ func ProcessReport(msg *kafka.Message, _ *kafka.Consumer) {
 							IntervalEnd:   interval_end_time,
 							UsageMetrics:  container_usage_metrics,
 						}
-						if err := workload_metric.CreateWorkloadMetrics(); err != nil {
-							log.Errorf("unable to add record to workload_metrics table: %v. Error: %v", workload_metric, err)
-							continue
-						}
+						workload_metric_arr = append(workload_metric_arr, workload_metric)
 					}
 
 				}
+				if err := model.BatchInsertWorkloadMetrics(workload_metric_arr, rh_account.OrgId); err != nil {
+					log.Errorf("unable to batch insert to workload_metrics table. Error: %v", err)
```

Comment:
@patilsuraj767 Any thoughts on completing the transaction here i.e. after we know there was an error with BatchInsertWorkloadMetrics?

Reply:
Not exactly sure what you mean by completing transaction here. DB transaction takes place inside the model i.e. inside `BatchInsertWorkloadMetrics` function.

Reply:
I see, I am suggesting something like below, in case there is an db insert error, where ever we are writing to DB.
https://github.com/RedHatInsights/ros-ocp-backend/blob/main/internal/services/recommendation_poller.go#L44

Reply:
This might require some more code, but we will be better off in the long run since bulk insert might prove to be expensive.

Reply:
As discussed internally in the meeting above suggestion are good to implement but as in future we will be moving to Thanos for storing the workload metrics and currently ROSOCP itself does not use workload_metrics data. So we are not going ahead and implementing the db transaction logic.